staging/src/k8s.io/cloud-provider/fake/fake.go

```diff
@@ -321,13 +322,20 @@ func (f *Cloud) InstanceMetadata(ctx context.Context, node *v1.Node) (*cloudprov
 	f.addCall("instance-metadata-by-provider-id")
 	f.addressesMux.Lock()
 	defer f.addressesMux.Unlock()
+
+	providerID := node.Spec.ProviderID
+	id, ok := f.ExtID[types.NodeName(node.Name)]
+	if providerID == "" && ok {
+		providerID = "fake://" + id
```

Comment:
On second thought, I think we want `Cloud` to have a field `ProviderID` that holds the value that would be returned here. This way you can define the fake provider ID from the test instead of it being opaque. What do you think?